app/views/pensions/unauthorisedPayments/UnauthorisedPaymentsCYAView.scala.html

```diff
@@ -0,0 +1,118 @@
+}
+
+@amountSurcharged = @{
+    rowWithContent("unauthorisedPayments.cya.amountSurcharged", if (cya.surchargeAmount.getOrElse(0) == 0) "" else cya.surchargeAmount.getOrElse(false), appConfig.tailoringEnabled, changeLink = controllers.pensions.unauthorisedPayments.routes.SurchargeAmountController.show(taxYear))
+}
+
+@nonUkTaxAmountSurcharged = @{
+    rowWithContent("unauthorisedPayments.cya.nonUkTaxAmountSurcharged", if (cya.surchargeTaxAmountQuestion == None) "" else if (cya.surchargeTaxAmountQuestion.get == false) "£0" else if (cya.surchargeTaxAmount.getOrElse(0) == 0) "" else cya.surchargeTaxAmount.getOrElse(0), appConfig.tailoringEnabled, changeLink = controllers.pensions.unauthorisedPayments.routes.DidYouPayNonUkTaxController.show(taxYear))
+}
+
+@amountNotSurcharged = @{
+    rowWithContent("unauthorisedPayments.cya.amountNotSurcharged", if (cya.noSurchargeAmount.getOrElse(0) == 0) "" else cya.noSurchargeAmount.getOrElse(false), appConfig.tailoringEnabled, changeLink = controllers.pensions.unauthorisedPayments.routes.NoSurchargeAmountController.show(taxYear))
+}
+
+@nonUkTaxAmountNotSurcharged = @{
+    rowWithContent("unauthorisedPayments.cya.nonUkTaxAmountNotSurcharged", if (cya.noSurchargeTaxAmountQuestion == None) "" else if (cya.noSurchargeTaxAmountQuestion.get == false) "£0" else if (cya.noSurchargeTaxAmount.getOrElse(0) == 0) "" else cya.noSurchargeTaxAmount.getOrElse(0), appConfig.tailoringEnabled, changeLink = controllers.pensions.unauthorisedPayments.routes.NonUkTaxOnAmountNotSurchargeController.show(taxYear))
+}
+
+@ukPensionSchemes = @{
+rowWithContent("unauthorisedPayments.common.ukPensionSchemes", cya.ukPensionSchemesQuestion.getOrElse(false), appConfig.tailoringEnabled, changeLink = controllers.pensions.unauthorisedPayments.routes.WhereAnyOfTheUnauthorisedPaymentsController.show(taxYear))
+}
+
+@summaryRows = @{
+    (Nil :+ unauthorisedPayments :+ amountSurcharged :+ nonUkTaxAmountSurcharged :+ amountNotSurcharged :+ nonUkTaxAmountNotSurcharged :+ ukPensionSchemes)
+}
+
+@surchargeQuestionTrueRows = @{
+    (Nil :+ unauthorisedPayments :+ amountSurcharged :+ nonUkTaxAmountSurcharged :+ ukPensionSchemes)
+}
+
+@noSurchargeQuestionTrueRows = @{
+     (Nil :+ unauthorisedPayments :+ amountNotSurcharged :+ nonUkTaxAmountNotSurcharged :+ ukPensionSchemes)
+}
+
+@unAuthorisedPaymentOnlyRow = @{
+     (Nil :+ rowWithContent("unauthorisedPayments.common.title", "No", appConfig.tailoringEnabled, changeLink = controllers.pensions.unauthorisedPayments.routes.UnAuthorisedPaymentsController.show(taxYear)))
+}
+
+@layout(pageTitle = headingForUse, taxYear = Some(taxYear)) {
+
+    @heading(headingForUse, Some(messages("common.unauthorisedPaymentsFromPension.caption", (taxYear - 1).toString, taxYear.toString)), "govuk-!-margin-bottom-5")
+
+    @summaryList((cya.surchargeQuestion.getOrElse(false), cya.noSurchargeQuestion.getOrElse(false), cya.ukPensionSchemesQuestion.getOrElse(false)) match {
```

Comment:
This can be moved into a separate object or class whereby all responsibility is passed to a function. That would allow for a unit testable component.